src/transformers/modeling_tf_bart.py

```diff
@@ -0,0 +1,1215 @@
+
+        self.dropout = config.dropout
+        self.layerdrop = config.encoder_layerdrop
+        self.output_hidden_states = config.output_hidden_states
+        self.output_attentions = config.output_attentions
+
+        embed_dim = embed_tokens.vocab_size
+        self.embed_scale = math.sqrt(embed_dim) if config.scale_embedding else 1.0
+        self.padding_idx = config.pad_token_id
+        self.max_source_positions = config.max_position_embeddings
+
+        self.embed_tokens = embed_tokens
+        self.embed_positions = TFLearnedPositionalEmbedding(
+            config.max_position_embeddings,
+            embed_tokens.hidden_size,
+            self.padding_idx,
+            config.extra_pos_embeddings,
+            name="embed_positions",
+        )
+        self.layers = [TFEncoderLayer(config, name=f"layers.{i}") for i in range(config.encoder_layers)]
+        self.layernorm_embedding = tf.keras.layers.LayerNormalization(epsilon=1e-5, name="layernorm_embedding")
+        self.layer_norm = (
+            tf.keras.layers.LayerNormalization(epsilon=1e-5, name="layer_norm")
+            if config.add_final_layer_norm
+            else None
+        )
+        self.return_dict = config.return_dict
+
+    def call(
+        self,
+        input_ids=None,
+        attention_mask=None,
+        output_attentions=False,
+        output_hidden_states=False,
+        return_dict=None,
+        training=False,
+    ):
+        """
+        Args:
+            input_ids (Tensor): tokens in the source language of shape
+                `(batch, src_len)`
+            attention_mask (Tensor): indicating which indices are padding tokens.
+        Returns:
+            namedtuple:
+                - **x** (Tensor): the last encoder layer's output of
+                  shape `(src_len, batch, embed_dim)`
+
+                - **encoder_states** (List[Tensor]): all intermediate
+                  hidden states of shape `(src_len, batch, embed_dim)`.
+                  Only populated if *return_all_hiddens* is True.
+                - **all_attentions** (List[Tensor]): Attention weights for each layer.
+                During training might not be of length n_layers because of layer dropout.
+        """
+        output_attentions = output_attentions if output_attentions is not None else self.output_attentions
+        output_hidden_states = output_hidden_states if output_hidden_states is not None else self.output_hidden_states
+        return_dict = return_dict if return_dict is not None else self.return_dict
+
+        # check attention mask and invert
+        if attention_mask is not None:
+            assert attention_mask._rank() == 2
+            attention_mask = tf.cast(attention_mask, dtype=tf.float32)
+            attention_mask = (1.0 - attention_mask) * LARGE_NEGATIVE
+        inputs_embeds = self.embed_tokens(input_ids)
+        embed_pos = self.embed_positions(input_ids)
+        x = inputs_embeds + embed_pos
```

Comment:
Not a big fan of `x` variable names => think `hidden_states` is better, but this is more a nit

Reply:
gunna leave it to be consistent with bart+cause I disagree. It's canonical at this point.